File: 1544-count-good-nodes-in-binary-tree/count-good-nodes-in-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    
    public int goodNodes(TreeNode root) {
        return countNodes( root, root.val, new int[]{0});
    }
    private int countNodes(TreeNode node , int maximum, int[] count){
        if(node==null) return 0;
        if(node.val>= maximum){
            count[0]+=1;
            maximum=node.val;
        }
        int left= countNodes(node.left,maximum,count);
        int right= countNodes(node.right,maximum,count);

        return count[0];
    }


}

File: 1544-count-good-nodes-in-binary-tree/count-good-nodes-in-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def goodNodes(self, root: TreeNode) -> int:

      
        self.res=[]
        def countGood(root,maxi):
            if root is None:
                return 
            if root.val>= maxi :
                self.res.append(root)
            maxi=max(root.val,maxi)
            countGood(root.left, maxi)
            countGood(root.right, maxi)
        countGood(root,float('-inf'))
        return len(self.res)
        

            

        